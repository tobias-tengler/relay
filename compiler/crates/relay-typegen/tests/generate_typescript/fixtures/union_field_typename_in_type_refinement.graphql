fragment a on Query {
  maybeNode {
    ... on NonNode {
      name
    }
    ... on Story {
      __typename
      actorCount
    }
  }
}