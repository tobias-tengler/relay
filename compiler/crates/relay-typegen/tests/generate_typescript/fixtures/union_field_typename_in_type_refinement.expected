==================================== INPUT ====================================
fragment a on Query {
  maybeNode {
    ... on NonNode {
      name
    }
    ... on Story {
      __typename
      actorCount
    }
  }
}
==================================== OUTPUT ===================================
import { FragmentRefs } from "relay-runtime";
export type a$data = {
  readonly maybeNode: {
    readonly __typename: "Story";
    readonly actorCount?: number | null | undefined;
    readonly name?: string | null | undefined;
  } | null | undefined;
  readonly " $fragmentType": "a";
};
export type a$key = {
  readonly " $data"?: a$data;
  readonly " $fragmentSpreads": FragmentRefs<"a">;
};
